### DISCUSSION TITLE
HQX algorithms using libvips

### DISCUSSION THREAD

**Discussion by shree220** [libvips/libvips#2020] [Ideas]
Is it possible to do pixel scaling using [HQX](https://en.wikipedia.org/wiki/Hqx) algorithms?

**Comment by jcupitt** [comment]
Hello, someone would need to implement it. Let's tag this as an enhancement.

**Comment by shree220** [comment]
There are some C/C++ implementations in GitHub:
LGPL License: https://github.com/grom358/hqx (Includes hq2x, hq3x and hq4x)
LGPL License: https://github.com/jflemer/hqx (Includes hq2x, hq3x and hq4x)
Apache License: https://github.com/brunexgeek/hqx (Includes hq2x and hq3x)

I wonder if one of these implementations could be adapted.
